rpc_state_reader_sn_api/src/lib.rs

```diff
@@ -356,13 +358,13 @@ impl RpcState {
 
         match result["type"].as_str().unwrap() {
```

Comment:
This should use `deserialize_transaction_json` instead of  repeating code

Reply:
done!